addons/mrp/models/product.py

```diff
@@ -56,14 +56,10 @@ class ProductProduct(models.Model):
     mo_count = fields.Integer('# Manufacturing Orders', compute='_compute_mo_count')
 
     def _compute_bom_count(self):
-        read_group_res = self.env['mrp.bom'].read_group([('product_id', 'in', self.ids)], ['product_id'], ['product_id'])
```

Comment:
Indeed, this search is incorrect as the product_id on the BoM is very optional.  (<-> product_tmpl_id)  A search/ readgroup on product_tmpl_id would be much simpler.  For perf, we could do read_group on product_tmpl_id with domain product_id = False and then do this one. (sum them afterwards)  Because this could be slowing down stuff.

Reply:
Might be enough, indeed

Reply:
Note that the field is only used in the form view, so the performance should not be an issue.